### PR TITLE
Provide mechanism to pin plugins

### PR DESCRIPTION
**Pull request by greg-hellings**
Provides a method to pin arbitrary Jenkins plugins

Addresses issue #99

### CODE REVIEW

**Comment by greg-hellings** [comment]
@scoheb Please also check this out

**Review by robled — APPROVED** [review]

File: cinch/roles/jenkins_master/defaults/main.yml [hunk 1]
```diff
@@ -32,6 +32,10 @@ jenkins_version: "1.651.3"
 # The list of Jenkins plugins that should be installed by default from the
 # configured Update Centers.
 jenkins_plugins: "{{ lookup('file', 'files/jenkins-plugin-lists/default.txt').split('\n') }}"
+# A list of Jenkins plugins that should not be upgraded, but should, instead, be
```
Comment (scoheb) [inline cinch/roles/jenkins_master/defaults/main.yml]: I would change it to be:

> Bundled plugins are ones that are included by default with Jenkins. The process of pinning a bundled plugin tells Jenkins to not overwrite the correct version of plugin with an included version during a Jenkins upgrade but rather use the one in JENKINS_HOME. Make this an array of the names of the plugins that should be pinned. NOTE: This only works for Jenkins 1.X

**Comment by scoheb** [comment]
Where do we mention the order of when pinning should happen?

**Comment by greg-hellings** [comment]
We don't need to mention it, we just put it at the correct location in the code. I've included it before the installation of the plugins, as you indicated.

**Pull request merged**